horovod/common/ops/rocm/hip_kernels.cu

```diff
@@ -0,0 +1,325 @@
+// Copyright (C) 2020 NVIDIA CORPORATION. All rights reserved.
```

Comment:
This file is now mostly identical to the previously existing `ops/cuda/cuda_kernels.cu`. As far as I can tell it only differs by short string replacements like `cuda` -> `hip`. So I guess leaving in the NVIDIA copyright header is a good way to go.

(The same holds for `hip_kernels.h` vs. `cuda_kernels.h`)

Reply:
I agree with the licence header. Given that both files are so identical, should there be a respective comment being added to cuda and hip versions to point to each other so developers changing one file know they should also change the other so they stay in sync?

Should we add something to CI to flag when they get out of sync?
Could both files be generated from one source file during compilation to ensure both are identical?

Reply:
It is a good idea to check if either is out of date. I will set it up in a late PR. Thank you.